src/udfs/optical_character_recognition.py

```diff
@@ -0,0 +1,115 @@
+
+import pandas as pd
+import numpy as np
+import easyocr
+
+
+from typing import List
+from src.models.catalog.frame_info import FrameInfo
+from src.models.catalog.properties import ColorSpace
+from src.udfs.abstract_udfs import AbstractClassifierUDF
+from src.udfs.gpu_compatible import GPUCompatible
+
+
+class OpticalCharacterRecognition(AbstractClassifierUDF, GPUCompatible):
+    """
+    Arguments:
+        threshold (float): Threshold for classifier confidence score
+
+    """
+
+    @property
+    def name(self) -> str:
+        return "opticalcharacterrecognition"
+
+    def to_device(self, device: str):
+        """
+
+        :param device:
+        :return:
+        """
+        self.model = easyocr.Reader(['en'], gpu = "cuda:{}".format(device))
+        return self
+    
+    
+    def __init__(self, threshold=0.85):
+        super().__init__()
+        self.threshold = threshold
+        self.model = easyocr.Reader(['en'])
+
+
+    @property
+    def input_format(self) -> FrameInfo:
+        return FrameInfo(-1, -1, 3, ColorSpace.RGB)
+
+
+    @property
+    def labels(self) -> List[str]:
+        """
+            Empty as there are no labels required for
+            optical character recognition
+        """
+        return
+
+    def classify(self, frames: np.ndarray) -> pd.DataFrame:
+        """
+        Performs predictions on input frames
+        Arguments:
+            frames (tensor): Frames on which OCR needs
+            to be performed
+
+        Returns:
+            tuple containing predicted_classes (List[List[str]]),
+            predicted_boxes (List[List[BoundingBox]]),
+            predicted_scores (List[List[float]])
+
+        """
+        
+        frames_list = frames.values.tolist()
+        frames = np.array(frames_list)
+
+        outcome = pd.DataFrame()
+        
+        for i in range(frames.shape[0]):
+            prediction = self.model.readtext_batched(frames[i])
```

Comment:
Isn't this logic processing one frame at a time? How about we call `self.model.readtext_batched(frames)`?

Reply:
I think there was some confusion due to the image size. I have made a few changes to make prediction on a batch of frames. Once we have the results for the entire batch, we traverse through each of the results and append them to the output dataframe. Creating a new PR for this!